app/api/tickets.py

```diff
@@ -241,6 +241,12 @@ def before_update_object(self, ticket, data, view_kwargs):
                     {'event_id': ticket.event.id},
                     "Event having paid ticket must have a payment method",
                 )
+        else:
+            if (data.get('deleted_at') and ticket.has_order_tickets):
```

Comment:
It should only check completed, placed, pending and initializing tickets

Reply:
expired, cancelled orders may prevent deletion this way which will create issues like this https://github.com/fossasia/open-event-frontend/issues/5430

Reply:
Create a new function as this will not just check if the ticket has orders. Name it `has_current orders` and use 
```python
Order.query.join(TicketHolder)
    .filter(TicketHolder.ticket_id == self.id, 
        or_(Order.status == 'completed',
            Order.status == 'placed',
            Order.status == 'pending',
            Order.status == 'initializing'))
```

Don't fetch all the orders like done above. Neither use `get_count`. Use `exists` query like here

https://github.com/fossasia/open-event-server/blob/d3582cbae6b65044baf9178c53f0885bfa11623b/app/models/video_stream.py#L41-L46